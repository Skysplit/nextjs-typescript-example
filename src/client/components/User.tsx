import React, { SFC } from 'react';
import { Link } from '@app/next/routes';

export type UserProps = {
  id: number,
  name: string,
};

const User: SFC<UserProps> = props => (
  <div>
    <hr/>
    <h3>{props.name}</h3>
    <Link route="users.view" params={{ id: props.id }}>
      <a>View profile</a>
    </Link>
  </div>
);

export default User;
