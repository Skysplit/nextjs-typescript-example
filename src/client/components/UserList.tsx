import React, { Fragment, SFC } from 'react';
import User, { UserProps } from './User';

export type UserListProps = {
  users: UserProps[],
};

const UserList: SFC<UserListProps> = props => (
  <Fragment>
    {props.users.map(user => (
      <User key={user.id} {...user} />
    ))}
  </Fragment>
);

export default UserList;
