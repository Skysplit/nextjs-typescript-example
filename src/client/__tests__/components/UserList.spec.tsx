import React from 'react';
import { shallow } from 'enzyme';
import UserList from '@app/client/components/UserList';
import { UserProps } from '@app/client/components/User';

describe('<UserList />', () => {
  test('test stub', () => {
    const users: UserProps[] = [
      { id: 1, name: 'foo' },
      { id: 2, name: 'bar' },
    ];
    const component = shallow(<UserList users={users} />);
    expect(component).toMatchSnapshot();
  });
});
