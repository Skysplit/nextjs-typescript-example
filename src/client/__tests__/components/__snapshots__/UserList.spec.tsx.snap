// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList /> test stub 1`] = `
<React.Fragment>
  <User
    id={1}
    key="1"
    name="foo"
  />
  <User
    id={2}
    key="2"
    name="bar"
  />
</React.Fragment>
`;

exports[`UserList test stub 1`] = `
<React.Fragment>
  <User
    id={1}
    key="1"
    name="foo"
  />
  <User
    id={2}
    key="2"
    name="bar"
  />
</React.Fragment>
`;
